data-pipeline/src/trace_exporter.rs

```diff
@@ -310,23 +457,110 @@ impl TraceExporterBuilder {
     }
 
-    pub fn build(mut self) -> anyhow::Result<TraceExporter> {
+    /// Set the header indicating the tracer has computed the top-level tag
+    pub fn set_client_computed_top_level(mut self) -> Self {
+        self.client_computed_top_level = true;
+        self
+    }
+
+    /// Set the header indicating the tracer has already computed stats.
+    /// This should not be used when stats computation is enabled.
+    pub fn set_client_computed_stats(mut self) -> Self {
+        self.client_computed_stats = true;
+        self
+    }
+
+    /// Enable stats computation on traces sent through this exporter
+    pub fn enable_stats(mut self, bucket_size: time::Duration) -> Self {
+        self.stats_bucket_size = Some(bucket_size);
+        self
+    }
+
+    /// Enable peer tags aggregation for stats computation (requires stats computation to be
+    /// enabled)
+    pub fn enable_stats_peer_tags_aggregation(mut self, peer_tags: Vec<String>) -> Self {
+        self.peer_tags_aggregation = true;
+        self.peer_tags = peer_tags;
+        self
+    }
+
+    /// Enable stats eligibility by span kind (requires stats computation to be
+    /// enabled)
+    pub fn enable_compute_stats_by_span_kind(mut self) -> Self {
+        self.compute_stats_by_span_kind = true;
+        self
+    }
+
+    pub fn build(self) -> anyhow::Result<TraceExporter> {
         let runtime = tokio::runtime::Builder::new_current_thread()
             .enable_all()
             .build()?;
 
+        let mut stats = StatsComputationStatus::StatsDisabled;
+
+        // Proxy mode does not support stats
+        if self.input_format != TraceExporterInputFormat::Proxy {
+            if let Some(bucket_size) = self.stats_bucket_size {
+                let stats_concentrator = Arc::new(Mutex::new(SpanConcentrator::new(
+                    bucket_size,
+                    time::SystemTime::now(),
+                    self.peer_tags_aggregation,
+                    self.compute_stats_by_span_kind,
+                    self.peer_tags,
+                )));
+
+                let cancellation_token = CancellationToken::new();
+
+                let mut stats_exporter = stats_exporter::StatsExporter::new(
+                    self.stats_bucket_size.unwrap(),
+                    stats_concentrator.clone(),
+                    stats_exporter::LibraryMetadata {
+                        hostname: self.hostname,
+                        env: self.env,
+                        version: self.version,
+                        lang: self.language.clone(),
+                        tracer_version: self.tracer_version.clone(),
+                        runtime_id: uuid::Uuid::new_v4().to_string(),
+                        service: self.service,
+                        ..Default::default()
+                    },
+                    Endpoint::from_url(stats_exporter::stats_url_from_agent_url(
+                        self.url.as_deref().unwrap_or("http://127.0.0.1:8126"),
```

Comment:
it seems this string is duplicated in at least one other place, should we make a constant for it? (or is that hard to do because of the borrow checker? :( )

Reply:
Most of the tracers define a default for the agent URL so it seems reasonable to me here? (Although I think _most_ default to UDS if the socket exists before trying HTTP?)